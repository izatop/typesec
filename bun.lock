{
  "lockfileVersion": 1,
  "workspaces": {
    "": {
      "name": "typesec",
      "devDependencies": {
        "@prettier/plugin-oxc": "^0.0.4",
        "@types/bun": "^1.2.20",
        "oxlint": "^1.12.0",
        "prettier": "^3.6.2",
        "typescript": "^5.9.2",
      },
    },
    "packages/bootstrap": {
      "name": "@typesec/bootstrap",
      "version": "0.1.0",
      "bin": {
        "typesec-sync": "bin/typesec-sync",
      },
    },
    "packages/cli": {
      "name": "@typesec/cli",
      "version": "0.1.0",
      "bin": {
        "typesec-cli": "bin",
      },
      "dependencies": {
        "@commander-js/extra-typings": "^14.0.0",
        "@types/bun": "^1.2.20",
        "@typesec/core": "workspace:*",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
        "@typesec/unit": "workspace:*",
        "commander": "^14.0.0",
        "radash": "^12.1.1",
      },
    },
    "packages/core": {
      "name": "@typesec/core",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
        "radash": "^12.1.1",
      },
    },
    "packages/graph": {
      "name": "@typesec/graph",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@typesec/core": "workspace:*",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
      },
    },
    "packages/gup": {
      "name": "@typesec/gup",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@typesec/core": "workspace:*",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
      },
    },
    "packages/serve": {
      "name": "@typesec/serve",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@typesec/core": "workspace:*",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
        "@typesec/unit": "workspace:*",
        "valibot": "^1.1.0",
      },
    },
    "packages/the": {
      "name": "@typesec/the",
      "version": "0.1.0",
    },
    "packages/tracer": {
      "name": "@typesec/tracer",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@types/chalk": "^2.2.4",
        "@typesec/the": "workspace:*",
        "chalk": "^5.6.0",
        "radash": "^12.1.1",
      },
    },
    "packages/unit": {
      "name": "@typesec/unit",
      "version": "0.1.0",
      "dependencies": {
        "@types/bun": "^1.2.20",
        "@typesec/the": "workspace:*",
        "@typesec/tracer": "workspace:*",
        "radash": "^12.1.1",
      },
    },
  },
  "packages": {
    "@commander-js/extra-typings": ["@commander-js/extra-typings@14.0.0", "", { "peerDependencies": { "commander": "~14.0.0" } }, "sha512-hIn0ncNaJRLkZrxBIp5AsW/eXEHNKYQBh0aPdoUqNgD+Io3NIykQqpKFyKcuasZhicGaEZJX/JBSIkZ4e5x8Dg=="],

    "@emnapi/core": ["@emnapi/core@1.4.5", "", { "dependencies": { "@emnapi/wasi-threads": "1.0.4", "tslib": "^2.4.0" } }, "sha512-XsLw1dEOpkSX/WucdqUhPWP7hDxSvZiY+fsUC14h+FtQ2Ifni4znbBt8punRX+Uj2JG/uDb8nEHVKvrVlvdZ5Q=="],

    "@emnapi/runtime": ["@emnapi/runtime@1.4.5", "", { "dependencies": { "tslib": "^2.4.0" } }, "sha512-++LApOtY0pEEz1zrd9vy1/zXVaVJJ/EbAF3u0fXIzPJEDtnITsBGbbK0EkM72amhl/R5b+5xx0Y/QhcVOpuulg=="],

    "@emnapi/wasi-threads": ["@emnapi/wasi-threads@1.0.4", "", { "dependencies": { "tslib": "^2.4.0" } }, "sha512-PJR+bOmMOPH8AtcTGAyYNiuJ3/Fcoj2XN/gBEWzDIKh254XO+mM9XoXHk5GNEhodxeMznbg7BlRojVbKN+gC6g=="],

    "@napi-rs/wasm-runtime": ["@napi-rs/wasm-runtime@0.2.12", "", { "dependencies": { "@emnapi/core": "^1.4.3", "@emnapi/runtime": "^1.4.3", "@tybys/wasm-util": "^0.10.0" } }, "sha512-ZVWUcfwY4E/yPitQJl481FjFo3K22D6qF0DuFH6Y/nbnE11GY5uguDxZMGXPQ8WQ0128MXQD7TnfHyK4oWoIJQ=="],

    "@oxc-parser/binding-android-arm64": ["@oxc-parser/binding-android-arm64@0.74.0", "", { "os": "android", "cpu": "arm64" }, "sha512-lgq8TJq22eyfojfa2jBFy2m66ckAo7iNRYDdyn9reXYA3I6Wx7tgGWVx1JAp1lO+aUiqdqP/uPlDaETL9tqRcg=="],

    "@oxc-parser/binding-darwin-arm64": ["@oxc-parser/binding-darwin-arm64@0.74.0", "", { "os": "darwin", "cpu": "arm64" }, "sha512-xbY/io/hkARggbpYEMFX6CwFzb7f4iS6WuBoBeZtdqRWfIEi7sm/uYWXfyVeB8uqOATvJ07WRFC2upI8PSI83g=="],

    "@oxc-parser/binding-darwin-x64": ["@oxc-parser/binding-darwin-x64@0.74.0", "", { "os": "darwin", "cpu": "x64" }, "sha512-FIj2gAGtFaW0Zk+TnGyenMUoRu1ju+kJ/h71D77xc1owOItbFZFGa+4WSVck1H8rTtceeJlK+kux+vCjGFCl9Q=="],

    "@oxc-parser/binding-freebsd-x64": ["@oxc-parser/binding-freebsd-x64@0.74.0", "", { "os": "freebsd", "cpu": "x64" }, "sha512-W1I+g5TJg0TRRMHgEWNWsTIfe782V3QuaPgZxnfPNmDMywYdtlzllzclBgaDq6qzvZCCQc/UhvNb37KWTCTj8A=="],

    "@oxc-parser/binding-linux-arm-gnueabihf": ["@oxc-parser/binding-linux-arm-gnueabihf@0.74.0", "", { "os": "linux", "cpu": "arm" }, "sha512-gxqkyRGApeVI8dgvJ19SYe59XASW3uVxF1YUgkE7peW/XIg5QRAOVTFKyTjI9acYuK1MF6OJHqx30cmxmZLtiQ=="],

    "@oxc-parser/binding-linux-arm-musleabihf": ["@oxc-parser/binding-linux-arm-musleabihf@0.74.0", "", { "os": "linux", "cpu": "arm" }, "sha512-jpnAUP4Fa93VdPPDzxxBguJmldj/Gpz7wTXKFzpAueqBMfZsy9KNC+0qT2uZ9HGUDMzNuKw0Se3bPCpL/gfD2Q=="],

    "@oxc-parser/binding-linux-arm64-gnu": ["@oxc-parser/binding-linux-arm64-gnu@0.74.0", "", { "os": "linux", "cpu": "arm64" }, "sha512-fcWyM7BNfCkHqIf3kll8fJctbR/PseL4RnS2isD9Y3FFBhp4efGAzhDaxIUK5GK7kIcFh1P+puIRig8WJ6IMVQ=="],

    "@oxc-parser/binding-linux-arm64-musl": ["@oxc-parser/binding-linux-arm64-musl@0.74.0", "", { "os": "linux", "cpu": "arm64" }, "sha512-AMY30z/C77HgiRRJX7YtVUaelKq1ex0aaj28XoJu4SCezdS8i0IftUNTtGS1UzGjGZB8zQz5SFwVy4dRu4GLwg=="],

    "@oxc-parser/binding-linux-riscv64-gnu": ["@oxc-parser/binding-linux-riscv64-gnu@0.74.0", "", { "os": "linux", "cpu": "none" }, "sha512-/RZAP24TgZo4vV/01TBlzRqs0R7E6xvatww4LnmZEBBulQBU/SkypDywfriFqWuFoa61WFXPV7sLcTjJGjim/w=="],

    "@oxc-parser/binding-linux-s390x-gnu": ["@oxc-parser/binding-linux-s390x-gnu@0.74.0", "", { "os": "linux", "cpu": "s390x" }, "sha512-620J1beNAlGSPBD+Msb3ptvrwxu04B8iULCH03zlf0JSLy/5sqlD6qBs0XUVkUJv1vbakUw1gfVnUQqv0UTuEg=="],

    "@oxc-parser/binding-linux-x64-gnu": ["@oxc-parser/binding-linux-x64-gnu@0.74.0", "", { "os": "linux", "cpu": "x64" }, "sha512-WBFgQmGtFnPNzHyLKbC1wkYGaRIBxXGofO0+hz1xrrkPgbxbJS1Ukva1EB8sPaVBBQ52Bdc2GjLSp721NWRvww=="],

    "@oxc-parser/binding-linux-x64-musl": ["@oxc-parser/binding-linux-x64-musl@0.74.0", "", { "os": "linux", "cpu": "x64" }, "sha512-y4mapxi0RGqlp3t6Sm+knJlAEqdKDYrEue2LlXOka/F2i4sRN0XhEMPiSOB3ppHmvK4I2zY2XBYTsX1Fel0fAg=="],

    "@oxc-parser/binding-wasm32-wasi": ["@oxc-parser/binding-wasm32-wasi@0.74.0", "", { "dependencies": { "@napi-rs/wasm-runtime": "^0.2.11" }, "cpu": "none" }, "sha512-yDS9bRDh5ymobiS2xBmjlrGdUuU61IZoJBaJC5fELdYT5LJNBXlbr3Yc6m2PWfRJwkH6Aq5fRvxAZ4wCbkGa8w=="],

    "@oxc-parser/binding-win32-arm64-msvc": ["@oxc-parser/binding-win32-arm64-msvc@0.74.0", "", { "os": "win32", "cpu": "arm64" }, "sha512-XFWY52Rfb4N5wEbMCTSBMxRkDLGbAI9CBSL24BIDywwDJMl31gHEVlmHdCDRoXAmanCI6gwbXYTrWe0HvXJ7Aw=="],

    "@oxc-parser/binding-win32-x64-msvc": ["@oxc-parser/binding-win32-x64-msvc@0.74.0", "", { "os": "win32", "cpu": "x64" }, "sha512-1D3x6iU2apLyfTQHygbdaNbX3nZaHu4yaXpD7ilYpoLo7f0MX0tUuoDrqJyJrVGqvyXgc0uz4yXz9tH9ZZhvvg=="],

    "@oxc-project/types": ["@oxc-project/types@0.74.0", "", {}, "sha512-KOw/RZrVlHGhCXh1RufBFF7Nuo7HdY5w1lRJukM/igIl6x9qtz8QycDvZdzb4qnHO7znrPyo2sJrFJK2eKHgfQ=="],

    "@oxlint-tsgolint/darwin-arm64": ["@oxlint-tsgolint/darwin-arm64@0.0.4", "", { "os": "darwin", "cpu": "arm64" }, "sha512-qL0zqIYdYrXl6ghTIHnhJkvyYy1eKz0P8YIEp59MjY3/zNiyk/gtyp8LkwZdqb9ezbcX9UDQhSuSO1wURJsq8g=="],

    "@oxlint-tsgolint/darwin-x64": ["@oxlint-tsgolint/darwin-x64@0.0.4", "", { "os": "darwin", "cpu": "x64" }, "sha512-c3nSjqmDSKzemChAEUv/zy2e9cwgkkO/7rz4Y447+8pSbeZNHi3RrNpVHdrKL/Qep4pt6nFZE+6PoczZxHNQjg=="],

    "@oxlint-tsgolint/linux-arm64": ["@oxlint-tsgolint/linux-arm64@0.0.4", "", { "os": "linux", "cpu": "arm64" }, "sha512-P2BA54c/Ej5AGkChH1/7zMd6PwZfa+jnw8juB/JWops+BX+lbhbbBHz0cYduDBgWYjRo4e3OVJOTskqcpuMfNw=="],

    "@oxlint-tsgolint/linux-x64": ["@oxlint-tsgolint/linux-x64@0.0.4", "", { "os": "linux", "cpu": "x64" }, "sha512-hbgLpnDNicPrbHOAQ9nNfLOSrUrdWANP/umR7P/cwCc1sv66eEs7bm4G3mrhRU8aXFBJmbhdNqiDSUkYYvHWJQ=="],

    "@oxlint-tsgolint/win32-arm64": ["@oxlint-tsgolint/win32-arm64@0.0.4", "", { "os": "win32", "cpu": "arm64" }, "sha512-ozKEppmwZhC5LMedClBEat6cXgBGUvxGOgsKK2ZZNE6zSScX7QbvJAOt3nWMGs8GQshHy/6ndMB33+uRloglQA=="],

    "@oxlint-tsgolint/win32-x64": ["@oxlint-tsgolint/win32-x64@0.0.4", "", { "os": "win32", "cpu": "x64" }, "sha512-gLfx+qogW21QcaRKFg6ARgra7tSPqyn+Ems3FgTUyxV4OpJYn7KsQroygxOWElqv6JUobtvHBrxdB6YhlvERbQ=="],

    "@oxlint/darwin-arm64": ["@oxlint/darwin-arm64@1.12.0", "", { "os": "darwin", "cpu": "arm64" }, "sha512-Pv+Ho1uq2ny8g2P6JgQpaIUF1FHPL32DfOlZhKqmzDT3PydtFvZp/7zNyJE3BIXeTOOOG1Eg12hjZHMLsWxyNw=="],

    "@oxlint/darwin-x64": ["@oxlint/darwin-x64@1.12.0", "", { "os": "darwin", "cpu": "x64" }, "sha512-kNXPH/7jXjX4pawrEWXQHOasOdOsrYKhskA1qYwLYcv/COVSoxOSElkQtQa+KxN5zzt3F02kBdWDndLpgJLbLQ=="],

    "@oxlint/linux-arm64-gnu": ["@oxlint/linux-arm64-gnu@1.12.0", "", { "os": "linux", "cpu": "arm64" }, "sha512-U7NETs02K55ZyDlgdhx4lWeFYbkUKcL+YcG+Ak70EyEt/BKIIVt4B84VdV1JzC71FErUipDYAwPJmxMREXr4Sg=="],

    "@oxlint/linux-arm64-musl": ["@oxlint/linux-arm64-musl@1.12.0", "", { "os": "linux", "cpu": "arm64" }, "sha512-e4Pb2eZu3V2BsiX4t4gyv9iJ8+KRT6bkoWM5uC9BLX7edsVchwLwL6LB2vPYusYdPPrxdjlFCg6ni+9wlw7FbQ=="],

    "@oxlint/linux-x64-gnu": ["@oxlint/linux-x64-gnu@1.12.0", "", { "os": "linux", "cpu": "x64" }, "sha512-qJK98Dj/z7Nbm0xoz0nCCMFGy0W/kLewPzOK5QENxuUoQQ6ymt7/75rXOuTwAZJ6JFTarqfSuMAA0pka6Tmytw=="],

    "@oxlint/linux-x64-musl": ["@oxlint/linux-x64-musl@1.12.0", "", { "os": "linux", "cpu": "x64" }, "sha512-jNeltpHc1eonSev/bWKipJ7FI6+Rc7EXh6Y7E0pm8e95sc1klFA29FFVs3FjMA6CCa+SRT0u0nnNTTAtf2QOiQ=="],

    "@oxlint/win32-arm64": ["@oxlint/win32-arm64@1.12.0", "", { "os": "win32", "cpu": "arm64" }, "sha512-T3fpNZJ3Q9YGgJTKc1YyvGoomSXnrV5mREz0QACE06zUzfS8EWyaYc/GN17FhHvQ4uQk/1xLgnM6FPsuLMeRhw=="],

    "@oxlint/win32-x64": ["@oxlint/win32-x64@1.12.0", "", { "os": "win32", "cpu": "x64" }, "sha512-2eC4XQ1SMM2z7bCDG+Ifrn5GrvP6fkL0FGi4ZwDCrx6fwb1byFrXgSUNIPiqiiqBBrFRMKlXzU9zD6IjuFlUOg=="],

    "@prettier/plugin-oxc": ["@prettier/plugin-oxc@0.0.4", "", { "dependencies": { "oxc-parser": "0.74.0" } }, "sha512-UGXe+g/rSRbglL0FOJiar+a+nUrst7KaFmsg05wYbKiInGWP6eAj/f8A2Uobgo5KxEtb2X10zeflNH6RK2xeIQ=="],

    "@tybys/wasm-util": ["@tybys/wasm-util@0.10.0", "", { "dependencies": { "tslib": "^2.4.0" } }, "sha512-VyyPYFlOMNylG45GoAe0xDoLwWuowvf92F9kySqzYh8vmYm7D2u4iUJKa1tOUpS70Ku13ASrOkS4ScXFsTaCNQ=="],

    "@types/bun": ["@types/bun@1.2.20", "", { "dependencies": { "bun-types": "1.2.20" } }, "sha512-dX3RGzQ8+KgmMw7CsW4xT5ITBSCrSbfHc36SNT31EOUg/LA9JWq0VDdEXDRSe1InVWpd2yLUM1FUF/kEOyTzYA=="],

    "@types/chalk": ["@types/chalk@2.2.4", "", { "dependencies": { "chalk": "*" } }, "sha512-pb/QoGqtCpH2famSp72qEsXkNzcErlVmiXlQ/ww+5AddD8TmmYS7EWg5T20YiNCAiTgs8pMf2G8SJG5h/ER1ZQ=="],

    "@types/node": ["@types/node@22.13.17", "", { "dependencies": { "undici-types": "~6.20.0" } }, "sha512-nAJuQXoyPj04uLgu+obZcSmsfOenUg6DxPKogeUy6yNCFwWaj5sBF8/G/pNo8EtBJjAfSVgfIlugR/BCOleO+g=="],

    "@types/react": ["@types/react@19.1.8", "", { "dependencies": { "csstype": "^3.0.2" } }, "sha512-AwAfQ2Wa5bCx9WP8nZL2uMZWod7J7/JSplxbTmBQ5ms6QpqNYm672H0Vu9ZVKVngQ+ii4R/byguVEUZQyeg44g=="],

    "@typesec/bootstrap": ["@typesec/bootstrap@workspace:packages/bootstrap"],

    "@typesec/cli": ["@typesec/cli@workspace:packages/cli"],

    "@typesec/core": ["@typesec/core@workspace:packages/core"],

    "@typesec/graph": ["@typesec/graph@workspace:packages/graph"],

    "@typesec/gup": ["@typesec/gup@workspace:packages/gup"],

    "@typesec/serve": ["@typesec/serve@workspace:packages/serve"],

    "@typesec/the": ["@typesec/the@workspace:packages/the"],

    "@typesec/tracer": ["@typesec/tracer@workspace:packages/tracer"],

    "@typesec/unit": ["@typesec/unit@workspace:packages/unit"],

    "bun-types": ["bun-types@1.2.20", "", { "dependencies": { "@types/node": "*" }, "peerDependencies": { "@types/react": "^19" } }, "sha512-pxTnQYOrKvdOwyiyd/7sMt9yFOenN004Y6O4lCcCUoKVej48FS5cvTw9geRaEcB9TsDZaJKAxPTVvi8tFsVuXA=="],

    "chalk": ["chalk@5.6.0", "", {}, "sha512-46QrSQFyVSEyYAgQ22hQ+zDa60YHA4fBstHmtSApj1Y5vKtG27fWowW03jCk5KcbXEWPZUIR894aARCA/G1kfQ=="],

    "commander": ["commander@14.0.0", "", {}, "sha512-2uM9rYjPvyq39NwLRqaiLtWHyDC1FvryJDa2ATTVims5YAS4PupsEQsDvP14FqhFr0P49CYDugi59xaxJlTXRA=="],

    "csstype": ["csstype@3.1.3", "", {}, "sha512-M1uQkMl8rQK/szD0LNhtqxIPLpimGm8sOBwU7lLnCpSbTyY3yeU1Vc7l4KT5zT4s/yOxHH5O7tIuuLOCnLADRw=="],

    "oxc-parser": ["oxc-parser@0.74.0", "", { "dependencies": { "@oxc-project/types": "^0.74.0" }, "optionalDependencies": { "@oxc-parser/binding-android-arm64": "0.74.0", "@oxc-parser/binding-darwin-arm64": "0.74.0", "@oxc-parser/binding-darwin-x64": "0.74.0", "@oxc-parser/binding-freebsd-x64": "0.74.0", "@oxc-parser/binding-linux-arm-gnueabihf": "0.74.0", "@oxc-parser/binding-linux-arm-musleabihf": "0.74.0", "@oxc-parser/binding-linux-arm64-gnu": "0.74.0", "@oxc-parser/binding-linux-arm64-musl": "0.74.0", "@oxc-parser/binding-linux-riscv64-gnu": "0.74.0", "@oxc-parser/binding-linux-s390x-gnu": "0.74.0", "@oxc-parser/binding-linux-x64-gnu": "0.74.0", "@oxc-parser/binding-linux-x64-musl": "0.74.0", "@oxc-parser/binding-wasm32-wasi": "0.74.0", "@oxc-parser/binding-win32-arm64-msvc": "0.74.0", "@oxc-parser/binding-win32-x64-msvc": "0.74.0" } }, "sha512-2tDN/ttU8WE6oFh8EzKNam7KE7ZXSG5uXmvX85iNzxdJfMssDWcj3gpYzZi1E04XuE7m3v1dVWl/8BE886vPGw=="],

    "oxlint": ["oxlint@1.12.0", "", { "optionalDependencies": { "@oxlint/darwin-arm64": "1.12.0", "@oxlint/darwin-x64": "1.12.0", "@oxlint/linux-arm64-gnu": "1.12.0", "@oxlint/linux-arm64-musl": "1.12.0", "@oxlint/linux-x64-gnu": "1.12.0", "@oxlint/linux-x64-musl": "1.12.0", "@oxlint/win32-arm64": "1.12.0", "@oxlint/win32-x64": "1.12.0", "oxlint-tsgolint": ">=0.0.1" }, "bin": { "oxlint": "bin/oxlint", "oxc_language_server": "bin/oxc_language_server" } }, "sha512-tBQ9aB00aYLlGXE21WJHnKQAI8xoi2V6Eiz/WvGV7FwU9YLYuNOurEEVbfoS5u0ODX8GLvGWj1fdHh5Rb74Kkw=="],

    "oxlint-tsgolint": ["oxlint-tsgolint@0.0.4", "", { "optionalDependencies": { "@oxlint-tsgolint/darwin-arm64": "0.0.4", "@oxlint-tsgolint/darwin-x64": "0.0.4", "@oxlint-tsgolint/linux-arm64": "0.0.4", "@oxlint-tsgolint/linux-x64": "0.0.4", "@oxlint-tsgolint/win32-arm64": "0.0.4", "@oxlint-tsgolint/win32-x64": "0.0.4" }, "bin": { "tsgolint": "bin/tsgolint.js" } }, "sha512-KFWVP+VU3ymgK/Dtuf6iRkqjo+aN42lS1YThY6JWlNi1GQqm7wtio/kAwssqDhm8kP+CVXbgZAtu1wgsK4XeTg=="],

    "prettier": ["prettier@3.6.2", "", { "bin": { "prettier": "bin/prettier.cjs" } }, "sha512-I7AIg5boAr5R0FFtJ6rCfD+LFsWHp81dolrFD8S79U9tb8Az2nGrJncnMSnys+bpQJfRUzqs9hnA81OAA3hCuQ=="],

    "radash": ["radash@12.1.1", "", {}, "sha512-h36JMxKRqrAxVD8201FrCpyeNuUY9Y5zZwujr20fFO77tpUtGa6EZzfKw/3WaiBX95fq7+MpsuMLNdSnORAwSA=="],

    "tslib": ["tslib@2.8.1", "", {}, "sha512-oJFu94HQb+KVduSUQL7wnpmqnfmLsOA/nAh6b6EH0wCEoK0/mPeXU6c3wKDV83MkOuHPRHtSXKKU99IBazS/2w=="],

    "typescript": ["typescript@5.9.2", "", { "bin": { "tsc": "bin/tsc", "tsserver": "bin/tsserver" } }, "sha512-CWBzXQrc/qOkhidw1OzBTQuYRbfyxDXJMVJ1XNwUHGROVmuaeiEm3OslpZ1RV96d7SKKjZKrSJu3+t/xlw3R9A=="],

    "undici-types": ["undici-types@6.20.0", "", {}, "sha512-Ny6QZ2Nju20vw1SRHe3d9jVu6gJ+4e3+MMpqu7pqE5HT6WsTSlce++GQmK5UXS8mzV8DSYHrQH+Xrf2jVcuKNg=="],

    "valibot": ["valibot@1.1.0", "", { "peerDependencies": { "typescript": ">=5" }, "optionalPeers": ["typescript"] }, "sha512-Nk8lX30Qhu+9txPYTwM0cFlWLdPFsFr6LblzqIySfbZph9+BFsAHsNvHOymEviUepeIW6KFHzpX8TKhbptBXXw=="],

    "@types/chalk/chalk": ["chalk@5.4.1", "", {}, "sha512-zgVZuo2WcZgfUEmsn6eO3kINexW8RAE4maiQ8QNs8CtpPCSyMiYsULR3HQYkm3w8FIA3SberyMJMSldGsW+U3w=="],
  }
}
