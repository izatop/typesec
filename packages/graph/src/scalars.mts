import {complex, primitive} from "./proto.mts";

export const string = primitive({
    name: "string",
    validate: (value): value is string => typeof value === "string",
});

export const boolean = primitive({
    name: "bool",
    validate: (value): value is boolean => typeof value === "boolean",
});

export const float = primitive({
    name: "float",
    validate: (value): value is number => typeof value === "number",
});

export const int = primitive({
    name: "int",
    validate: (value): value is number => typeof value === "number" && value % 1 === 0,
});

export const bigint = complex({
    name: "bigint",
    format: string(),
    validate: (value): value is bigint => typeof value === "bigint",
    deserialize: (value) => BigInt(value),
    serialize: (value) => value.toString(),
});

export const datetime = complex({
    name: "datetime",
    format: string(),
    validate: (value): value is Date => value instanceof Date && value.toString() !== "Invalid Date",
    deserialize: (value) => new Date(value),
    serialize: (value) => value.toISOString(),
});

export const scalars = {
    string,
    boolean,
    float,
    int,
    bigint,
    datetime,
};
