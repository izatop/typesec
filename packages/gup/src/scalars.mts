import {date} from "@typesec/the/date";
import {is} from "@typesec/the/fn";
import {numbers} from "@typesec/the/numbers";
import {codec, primitive, proto} from "./proto.mts";

const string = primitive({
    id: "string",
    kind: "string",
    isValid: (value): value is string => is(value, "string"),
});

const boolean = primitive({
    id: "boolean",
    kind: "boolean",
    isValid: (value): value is boolean => is(value, "boolean"),
});

const float = proto.primitive({
    id: "float",
    kind: "number",
    isValid: numbers.isFinite,
});

const int = proto.primitive({
    id: "int",
    kind: "number",
    isValid: numbers.isInt,
});

const bigint = codec({
    id: "bigint",
    isKind: (value): value is bigint => is(value, "bigint"),
    isValid: (value): value is bigint => is(value, "bigint"),
    encode: (value) => value.toString(),
    decode: (value) => BigInt(value),
});

const ISODate = proto.codec({
    id: "ISODate",
    isKind: date.is,
    isValid: date.valid,
    encode: (value) => value.toISOString(),
    decode: (value) => new Date(value),
});

export const scalars = {
    ISODate,
    boolean,
    string,
    bigint,
    float,
    int,
};
