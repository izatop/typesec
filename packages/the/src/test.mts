import type {Equal, Extends, HasNull, HasUndefined} from "./type.mjs";

export function isXEqualToY<X, Y>(equal: Equal<X, Y>): boolean {
    return equal;
}

export function isXExtendsOfY<X, Y>(equal: Extends<X, Y>): boolean {
    return equal;
}

export function isXHasNull<X>(equals: HasNull<X>) {
    return equals;
}

export function isXHasUndefined<X>(equals: HasUndefined<X>) {
    return equals;
}
