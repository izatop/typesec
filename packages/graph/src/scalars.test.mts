import type {Rec} from "@typesec/the";
import {describe, expect, test} from "bun:test";
import {scalars} from "./scalars.mts";

describe("scalars", async () => {
    test("scalars should work", () => {
        const samples: Rec<keyof typeof scalars, unknown> = {
            bigint: 1n,
            boolean: true,
            float: 1.1,
            int: 1,
            string: "",
            datetime: new Date(0),
        };

        const any = Object.values(samples);

        expect(
            Object.values(scalars)
                .map((scalar) => scalar())
                .map((type) => [type.name, any.map((value) => [value, type.validate(value)])]),
        ).toMatchSnapshot();
    });

    test("datetime should work", () => {
        const value = new Date(0);
        const type = scalars.datetime();
        expect(type.validate(value)).toBeTrue();
        expect(type.serialize(value)).toBe(value.toISOString());
        expect(type.deserialize(value.toISOString())).toEqual(value);
    });

    test("bigint should work", () => {
        const value = 1n;
        const type = scalars.bigint();
        expect(type.validate(value)).toBeTrue();
        expect(type.serialize(value)).toBe(value.toString());
        expect(type.deserialize(value.toString())).toBe(value);
    });
});
