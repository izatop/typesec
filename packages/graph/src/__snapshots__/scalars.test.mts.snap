// Bun Snapshot v1, https://bun.sh/docs/test/snapshots

exports[`scalars scalars should work 1`] = `
[
  [
    "string",
    [
      [
        1n,
        false,
      ],
      [
        true,
        false,
      ],
      [
        1.1,
        false,
      ],
      [
        1,
        false,
      ],
      [
        "",
        true,
      ],
      [
        1970-01-01T00:00:00.000Z,
        false,
      ],
    ],
  ],
  [
    "bool",
    [
      [
        1n,
        false,
      ],
      [
        true,
        true,
      ],
      [
        1.1,
        false,
      ],
      [
        1,
        false,
      ],
      [
        "",
        false,
      ],
      [
        1970-01-01T00:00:00.000Z,
        false,
      ],
    ],
  ],
  [
    "float",
    [
      [
        1n,
        false,
      ],
      [
        true,
        false,
      ],
      [
        1.1,
        true,
      ],
      [
        1,
        true,
      ],
      [
        "",
        false,
      ],
      [
        1970-01-01T00:00:00.000Z,
        false,
      ],
    ],
  ],
  [
    "int",
    [
      [
        1n,
        false,
      ],
      [
        true,
        false,
      ],
      [
        1.1,
        false,
      ],
      [
        1,
        true,
      ],
      [
        "",
        false,
      ],
      [
        1970-01-01T00:00:00.000Z,
        false,
      ],
    ],
  ],
  [
    "bigint",
    [
      [
        1n,
        true,
      ],
      [
        true,
        false,
      ],
      [
        1.1,
        false,
      ],
      [
        1,
        false,
      ],
      [
        "",
        false,
      ],
      [
        1970-01-01T00:00:00.000Z,
        false,
      ],
    ],
  ],
  [
    "datetime",
    [
      [
        1n,
        false,
      ],
      [
        true,
        false,
      ],
      [
        1.1,
        false,
      ],
      [
        1,
        false,
      ],
      [
        "",
        false,
      ],
      [
        1970-01-01T00:00:00.000Z,
        true,
      ],
    ],
  ],
]
`;
